sse2neon.h

```diff
@@ -7409,6 +7409,30 @@ FORCE_INLINE __m128i _mm_cvtepu8_epi64(__m128i a)
     return vreinterpretq_m128i_u64(u64x2);
 }
 
+// Conditionally multiply the packed double-precision (64-bit) floating-point
+// elements in a and b using the high 4 bits in imm8, sum the four products, and
+// conditionally store the sum in dst using the low 4 bits of imm8.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_dp_pd
+FORCE_INLINE __m128d _mm_dp_pd(__m128d a, __m128d b, const int imm)
+{
+    // Conditional multiplication
+    __m128d mul = _mm_mul_pd(a, b);
+    const __m128d mulMask = _mm_castsi128_pd(_mm_set_epi64x(
+        imm & 0x20 ? UINT64_MAX : 0, imm & 0x10 ? UINT64_MAX : 0));
```

Comment:
Since `imm` is constant, the use of ternary conditional operator (such as `imm & 0x20 ? UINT64_MAX : 0`) can be set at the beginning of function body.

Reply:
I make them declared as constant variables at the beginning of the function body.